engine/runtime/src/main/java/org/enso/interpreter/runtime/data/text/Text.java

```diff
@@ -193,19 +193,58 @@ String toDisplayString(
       boolean allowSideEffects,
       @Cached("build()") @Cached.Shared("strings") ToJavaStringNode toJavaStringNode) {
     String str = toJavaStringNode.execute(this);
-    // TODO This should be more extensible
-    String replaced =
-        str.replace("\\", "\\\\")
-            .replace("'", "\\'")
-            .replace("\n", "\\n")
-            .replace("\t", "\\t")
-            .replace("\u0007", "\\a")
-            .replace("\u0008", "\\b")
-            .replace("\u000c", "\\f")
-            .replace("\r", "\\r")
-            .replace("\u000B", "\\v")
-            .replace("\u001B", "\\e");
-    return "'" + replaced + "'";
+    int len = str.length();
+    int outputLength = len + 2; // Precise if there are no special characters.
+
+    // TODO This should be more extensible; while it's still a small fixed set,
+    // a switch is probably fastest (unconfirmed).
+
+    StringBuffer strBuf = new StringBuffer(outputLength);
+
+    strBuf.append('\'');
+
+    for (int i = 0; i < len; ++i) {
+      char c = str.charAt(i);
+      switch (c) {
```

Comment:
Done; I used the arrow syntax, but I didn't make it a switch expression, since the value it appends isn't always a string. In the default case it's a character, and I wouldn't want to slow down the common case by converting the char to a string